zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -338,6 +370,44 @@ impl<F: FieldExt> ExecutionConfig<F> {
         &self,
         layouter: &mut impl Layouter<F>,
         block: &Block<F>,
+    ) -> Result<(), Error> {
+        // TODO: Assign region upto the desired capacity
```

Comment:
Even we have already assinged all steps in a block, we still need to pad the circuit to match the capacity, which is decided at keygen, right?

Reply:
I thought that halo2 when building behind the polynomials that need to be committed will create them with a fixed key length.

So we don't actually need to do anything in particular. It's like if dummy constraints were added to pad the circuit to the KEY size choosen.

@therealyingtong anyway will know the correct behaviour. Will ping her to see if can provide some insight.

Anyway, don't take this as a blocker to merge! :)

Reply:
Are we discussing padding the circuit with dummy inputs that still pass the gates? I do think that will need to be done at production.

Reply:
We're discussing about the gadget that will contain all the tx block step's constraints. 

This has to have a maximum limit which is related to the `k` provided to the preprocessing. And we were wondering if we need to padd with dummy constraints when we have less steps than the maximum or we don't.

We say all of that because the number cannot be arbitrary. Otherwise, we will end up with different circuit descriptions for each block.

Reply:
> So we don't actually need to do anything in particular. It's like if dummy constraints were added to pad the circuit to the KEY size choosen.

In `halo2` we always need to enable the selector manually IIUC, and the comment is a reminder of that, once we have decided how to pad the leftover steps.

> This has to have a maximum limit which is related to the `k` provided to the preprocessing.

Yes, we can do that by passing `k` as input for this function, or with const generic for this whole gadget.